File: backend/components/metiers/administration_publique/tests/plugins/plugins.test.py
# plugins.test.py – Test d’intégration avancé Plugins (fusionné, import direct PluginManager)
from . import PluginManager

class DummyPlugin:
    def run(self, x):
        return x + 1

def test_plugin_manager_integration():
    pm = PluginManager()
    plugin = DummyPlugin()
    pm.register('dummy', plugin)
    assert pm.get('dummy') is plugin
    assert pm.execute('dummy', 2) == 3
    pm.unregister('dummy')
    assert pm.get('dummy') is None

File: backend/components/metiers/administration_publique/tests/plugins/plugins.test.js
// plugins.test.js – Test d’intégration avancé Plugins (fusionné, import direct PluginManager)
const { PluginManager } = require('./__init__');

describe('Plugins Threed – Intégration', () => {
  it('peut enregistrer, exécuter et désenregistrer un plugin', () => {
    class DummyPlugin { run(x) { return x + 1; } }
    const pm = new PluginManager();
    pm.register('dummy', new DummyPlugin());
    expect(pm.get('dummy')).toBeDefined();
    expect(pm.execute('dummy', 2)).toBe(3);
    pm.unregister('dummy');
    expect(pm.get('dummy')).toBeUndefined();
  });
});
